### DISCUSSION TITLE
Thank you

### DISCUSSION THREAD

**Discussion by anohren** [pannal/plex-for-kodi#174] [General]
Just wanted to say thanks for breathing life into this addon. I just found this incarnation so I haven't really used it yet, but it's fun just to be able to launch it (something the old official addon doesn't do anymore) and play a track.

**Comment by pannal** [comment]
Thanks for the praise! We've come a long way from the original, though :)

Apart from the still dated UX/UI, PM4K might be the most versatile Plex client available.

I hope you're trying out beta 0.8.0 and not the "stable" one from the official repository. Apart from fixes, watchlist integration is planned for 0.8.0, and maybe a more modern UX for 0.8.1 (hidden vertical menu, maybe optional).

**Reply by anohren** [reply]
Following the installation instructions in the readme left me with version 0.7.9-rev4 from the Don't Panic repo, but it appears to me that the official Kodi repo also serves version 0.7.9 (?) No newer version seems available from Don't Panic. I recall there being some differently named addons in the repo but the letter combinations didn't ring a bell at the time and their descriptions were identical so I went ahead with the instructions. 

Yes, I saw the feature request for modernizing the appearance, but since I have zero other installs of Plex clients (other than on iOS) I haven't much of a reference of what's outdated or undesirable about the current one. I just like the design because of the differences in flow compared to Kodi, and I find the layout much more sensible (e.g. player + playlist split screen). Still looks great to me. 

Since I've stayed away from set top boxes I don't know enough history about who came first with the general design (Apple TV, Plex, someone else...) Maybe parts of it feel worn out to older players than me 🤷‍♂️

I'm trying it out on the shittiest little 32 bit machine possible though (except when compared with other netbooks from 2009) so I guarantee I will stub my toe on every single corner.

**Reply by pannal** [reply]
You can get the latest beta with a lot of fixes and improvements here: https://forums.plex.tv/t/pm4k-plexmod-for-kodi-18-19-20-21-22/481208/3509

It also includes an auto updater service (set PM4K to beta channel in its settings after overwriting the one from the repo), so you only have to do this manual installation step once. 

> I recall there being some differently named addons

The one in the official Kodi repository is named PM4K for Plex due to copyright reasons. Using Don't Panic is the better way, though.

**Reply by anohren** [reply]
Thanks, I'll check it out. 

> > I recall there being some differently named addons
> 
> The one in the official Kodi repository is named PM4K for Plex due to copyright reasons

I must have confused it with the Plextuary skin — it was the one with the inexplicably named variants ("CE", "CPM")

**Reply by pannal** [reply]
CE = CoreELEC, CPM = CoreELEC cpm build

**Reply by anohren** [reply]
Thanks for the guidance. I did try both the release as well as the beta 8 on two devices. 

Unfortunately it seems to break down when I take away its internet connection, which doesn't work for my self-hosting goals. I'm used to it — even the official iOS Plex app barely manages to work even when giving it static local server addresses. 

Anyway, I then tried inputting the static ip address in settings, and that worked only when using 127.0.0.1 on the server machine itself. 

To get the networked client to work I had to give it internet access and then it worked within a few seconds, and that's after manually giving it the correct address. Then it continued to work without internet access 🤷‍♂️

I'm personally at a loss when manually addresses should work, in any client app (in iOS for instance it seems you need to be logged out for it to work — I can't make sense of it). That's a shame, I was hoping for an offline-enabled solution. 

I'm also getting stuck in infinite loading screens both in beta and release, e.g. when choosing to Exit the addon, forcing me to restart Kodi completely. I'll just assume it's because of syncing the UI control flow to some perpetual network call that never returns.

**Reply by anohren** [reply]
@pannal By the way, let me know if any of the above issues is valuable as a separate issue here, and I'll try to help to recreate it. 

I imagine that you might be aware of things already, some things are by design, the beta is moving fast, or you're planning to go in another direction, so I won't clutter the issue section with things that maybe aren't relevant.
